### PR TITLE
created Numberdrome function

### PR DESCRIPTION
**Pull request by vdimition**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+class Numberdrome {
```
Comment (ost-ptk) [inline main.js]: use strict mode

File: main.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  constructor() {
+    this.numbers = [];
+  }
+
+  addNumber(n) {
+    this.numbers.push(n);
+  }
+
+  removeNumber(n) {
+    const index = this.numbers.findIndex(num => num === n);
```
Comment (gestap) [inline main.js]: in this case, you can use `indexOf`

File: main.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    this.numbers.splice(index, 1);
+  }
+
+  sum() {
+    if (this.numbers.length) {
+      return this.numbers.reduce(
```
Comment (gestap) [inline main.js]: you can use initialValue in `reduce` instead of `if...else` construction

File: main.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      );
+    } else {
+      return 0;
+    }
+  }
+
+  product() {
+    if (this.numbers.length) {
+      return this.numbers.reduce(
+        (accumulator, currentValue) => accumulator * currentValue
```
Comment (gestap) [inline main.js]: you can use initialValue in `reduce` instead of `if...else` construction

File: main.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    if (this.numbers.length) {
+      let min = Infinity;
+      this.numbers.map(a => {
+        if (a < min) {
+          min = a;
+        }
+      })    
+      return min;
+    } else {
+      return console.log('error: there are no numbers');
```
Comment (gestap) [inline main.js]: you need to `throw an error` instead return undefined

File: main.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    if (this.numbers.length) {
+      let max = -Infinity;
+      this.numbers.map(a => {
+        if (a > max) {
+          max = a;
+        }
+      })    
+      return max;
+    } else {
+      return console.log('error: there are no numbers');
```
Comment (gestap) [inline main.js]: you need to `throw an error` instead return undefined

File: main.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        (accumulator, currentValue) => accumulator * currentValue
+      );
+    } else {
+      return 1;
+    }
+  }
+
+  min() {
+    if (this.numbers.length) {
+      let min = Infinity;
```
Comment (gestap) [inline main.js]: try to use `Math` method

File: main.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      })    
+      return min;
+    } else {
+      return console.log('error: there are no numbers');
+    }
+  }
+
+  max() {
+    if (this.numbers.length) {
+      let max = -Infinity;
```
Comment (gestap) [inline main.js]: try to use `Math` method

**Review by ost-ptk — CHANGES_REQUESTED** [review]

File: main.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
+"use strict";
+
+class Numberdrome {
+  constructor() {
+    this.numbers = [];
+  }
+
+  addNumber(n) {
+    this.numbers.push(n);
```
Comment (nicholas38466) [inline main.js]: потрібна перевірка на тип n

**Review by NclsVI — CHANGES_REQUESTED** [review]

**Review by NclsVI — APPROVED** [review]

**Review by NclsVI — APPROVED** [review]

**Pull request closed without merging**